src/relayserver/RelayServer.ts

```diff
@@ -429,31 +435,40 @@ export class RelayServer extends EventEmitter {
 
   async replenishWorker (workerIndex: number): Promise<void> {
     const workerAddress = this.getAddress(workerIndex)
-    const workerBalance = toBN(await this.contractInteractor.getBalance(workerAddress))
+    const workerBalance = await this.getWorkerBalance(workerIndex)
     if (workerBalance.lt(toBN(this.workerMinBalance))) {
       const refill = toBN(this.workerTargetBalance).sub(workerBalance)
       const balance = await this.getManagerBalance()
+      const managerHubBalance = await this.relayHubContract?.balanceOf(this.managerAddress) ?? toBN(0)
       console.log(
         `== replenishWorker(${workerIndex}): mgr balance=${balance.div(
-          toBN(1e18)).toString()} worker balance=${workerBalance.div(
+          toBN(1e18)).toString()}  manager hub balance=${managerHubBalance.div(toBN(1e18)).toString()} 
+          worker balance=${workerBalance.div(
           toBN(1e18)).toString()} refill=${refill.div(toBN(1e18)).toString()}`)
-      if (refill.lt(balance.sub(toBN(minimumRelayBalance)))) {
+      if (refill.lt(managerHubBalance.sub(toBN(minimumRelayBalance)))) {
+        const method = this.relayHubContract?.contract.methods.withdraw(toHex(refill), workerAddress)
+        await this._sendTransaction({
+          signerIndex: 0,
+          destination: this.relayHubContract?.address as string,
+          method
+        })
+      } else if (refill.lt(balance.sub(toBN(minimumRelayBalance)))) {
         await this._sendTransaction({
           signerIndex: 0,
           destination: workerAddress,
           value: toHex(refill),
           gasLimit: mintxgascost.toString()
         })
-        await this.getManagerBalance()
       } else {
-        console.log(
-          `== replenishWorker: can't replenish: mgr balance too low ${balance.div(toBN(1e18)).toString()} refill=${refill.div(
-            toBN(1e18)).toString()}`)
+        const message = `== replenishWorker: can't replenish: mgr balance too low ${balance.div(toBN(1e18)).toString()} refill=${refill.div(
+          toBN(1e18)).toString()}`
+        this.emit('fundingNeeded', message)
+        console.log(message)
       }
     }
   }
 
-  async _worker (blockHeader: BlockHeader): Promise<TransactionReceipt | void> {
+  async _worker (blockHeader: BlockHeader): Promise<TransactionReceipt[]> {
```

Comment:
I saw tests that use `_worker()`'s return value - but they assume its a single receipt.
how can they work?

Reply:
I changed it in this PR. No tests assume a single receipt.